patches/tModLoader/Terraria/Initializers/LaunchInitializer.cs.patch

```diff
@@ -22,6 +22,59 @@
  		}
  
  		private static void LoadClientParameters(Main game) {
+-			string iP;
+-			if ((iP = TryParameter("-j", "-join")) != null)
+-				game.AutoJoin(iP);
++			string ip;
++			if ((ip = TryParameter("-j", "-join")) != null) {
++
++				int port = -1;
++				if (ip.Contains(':')) {
++					int colonInd = ip.LastIndexOf(':');
++					if (int.TryParse(ip.Substring(colonInd + 1), out int portNum)) {
++						port = portNum;
++						ip = ip.Substring(0, colonInd);
++					}
++				}
```

Comment:
This is helpful, but why not put it in `Netplay.SetRemoteIP`

Reply:
So edit `SetRemoteIP` to accept port?

Reply:
Yes